test/blockNumber_test.js

```diff
@@ -1,12 +1,34 @@
 // @format
 import test from "ava";
+import esmock from "esmock";
+import fetchMock from "fetch-mock";
 
-import getBlockNumber from "../src/blockNumber.js";
-
-test("getting block number", async t => {
+test("getting block number", async (t) => {
   const options = {
     url: "https://cloudflare-eth.com",
   };
+
+  const { default: getBlockNumber } = await esmock(
+    "../src/blockNumber.js",
+    null,
+    {
+      "cross-fetch": {
+        default: fetchMock.sandbox().post(
+          {
+            url: options.url,
+            body: {
+              method: "eth_blockNumber",
+            },
+            matchPartialBody: true,
```

Comment:
Why is `matchPartialBody` needed here? Aren't we matching the body exactly?

Reply:
In some cases we are not matching the whole body as the it contained a lot of parameters which added a lot of lines into the test case. I'll check once again if we can match the whole body in this case.